wd_demo.au3

```diff
@@ -173,43 +173,63 @@ Func RunDemo($idDebugging, $idBrowsers, $idUpdate, $idHeadless)
 	; Execute browser setup routine for user's browser selection
 	Local $sCapabilities = Call($aBrowsers[_GUICtrlComboBox_GetCurSel($idBrowsers)][1], $bHeadless)
 
-	_WD_Startup()
-	If @error <> $_WD_ERROR_Success Then Return
+	ConsoleWrite("> _WD_Startup" & @CRLF)
+	Local $iWebDriver_PID = _WD_Startup()
+	If _RunDemo_ErrorHander((@error <> $_WD_ERROR_Success), @error, @extended, $iWebDriver_PID, $sSession) Then Return
 
+	ConsoleWrite("> _WD_CreateSession" & @CRLF)
 	$sSession = _WD_CreateSession($sCapabilities)
-
-	Local $iError
-	If @error = $_WD_ERROR_Success Then
-		For $iIndex = 0 To UBound($aDemoSuite, $UBOUND_ROWS) - 1
-			If $aDemoSuite[$iIndex][1] Then
-				ConsoleWrite("+Running: " & $aDemoSuite[$iIndex][0] & @CRLF)
-				If $aDemoSuite[$iIndex][2] Then
-					Call($aDemoSuite[$iIndex][0], $sBrowser)
-				Else
-					Call($aDemoSuite[$iIndex][0])
-				EndIf
-				$iError = @error
-				If $iError = $_WD_ERROR_UserAbort Then
-					ConsoleWrite("- Aborted: " & $aDemoSuite[$iIndex][0] & @CRLF)
-					ExitLoop
-				EndIf
-				ConsoleWrite("+Finished: " & $aDemoSuite[$iIndex][0] & @CRLF)
+	If _RunDemo_ErrorHander((@error <> $_WD_ERROR_Success), @error, @extended, $iWebDriver_PID, $sSession) Then Return
+
+	Local $iError = -1
+	For $iIndex = 0 To UBound($aDemoSuite, $UBOUND_ROWS) - 1
+		If $aDemoSuite[$iIndex][1] Then
+			ConsoleWrite("+ Running: " & $aDemoSuite[$iIndex][0] & @CRLF)
+			If $aDemoSuite[$iIndex][2] Then
+				Call($aDemoSuite[$iIndex][0], $sBrowser)
 			Else
-				ConsoleWrite("Bypass: " & $aDemoSuite[$iIndex][0] & @CRLF)
+				Call($aDemoSuite[$iIndex][0])
 			EndIf
-		Next
-	EndIf
-
-	If $iError = $_WD_ERROR_UserAbort Then
-		MsgBox($MB_ICONINFORMATION, 'Demo aborted!', 'Click "Ok" button to shutdown the browser and console')
-	Else
-		MsgBox($MB_ICONINFORMATION, 'Demo complete!', 'Click "Ok" button to shutdown the browser and console')
-	EndIf
+			$iError = @error
+			If $iError = $_WD_ERROR_UserAbort Then
```

Comment:
all this Error's `ConsoleWrite` could be moved to `_RunDemo_ErrorHander`

Reply:
Done: [ConsoleWrite move to _RunDemo_ErrorHander()](https://github.com/Danp2/WebDriver/pull/176/commits/7d30828f43757465c14cfb3b8fc6e8e156796411)

Reply:
How is this an improvement? It seems to me that it only adds complexity and forces you to lookup the name of the current running routine, which was already known in `$aDemoSuite[$iIndex][0]`. 🤔

Reply:
> How is this an improvement?

We are moving things to _RunDemo_ErrorHander() so I moved also this part.
It looks for me so natural to do.

Reply:
it also improves MsgBox displaying accurate DemoName

Reply:
I suggest that this would be a better alternative --

`_RunDemo_ErrorHander(True, $iError, @extended, $iWebDriver_PID, $sSession, $aDemoSuite[$iIndex][0])`

This way you avoid the UBound checks, etc.